functions-and-scope.js

```diff
@@ -0,0 +1,199 @@
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 6
+let outcome = 0
+
+for (let i = 0; i < grades.length; i++) {
+    if( grades[i] >= 8) {
+        outcome = outcome + 1
+    }
+}
+
+console.log(outcome)
+
+
+/*  1b: Omschrijven tot een herbruikbare functie   */
+// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+
+let cumlaudeOutcome = 0;
+function cumlaude (array){
+    for (let i = 0; i < array.length; i++) {
+        if( array[i] >= 8) {
+            cumlaudeOutcome = cumlaudeOutcome + 1
+        }
+    }
+    return cumlaudeOutcome
+}
+const outcome2 = cumlaude([8, 9, 4, 6, 10])
+console.log(outcome2)
+
+// ---- Verwachte uitkomsten:
+// cumLaude(grades) geeft 6
+// cumLaude([6, 4, 5]) geeft 0
+// cumLaude([8, 9, 4, 6, 10]) geeft 3
+
+
+/* Opdracht  2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven  */
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe wordt een gemiddelde berekend?
+
+//alles bij elkaar opgeteld en dan door elkaar gedeeld
+
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+
+//alle cijfers bij elkaar gedeeld door the length
+
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// Log het antwoord in de terminal.
+
+let averageMark = 0
+for (let i = 0; i < grades.length; i++) {
+    if ([i] < grades.length -1 ) {
+        averageMark = averageMark + grades[i]
+    } else {
+        averageMark = averageMark + grades[i]
+        averageMark =  averageMark / grades.length
+    }
+
+}
+
+console.log(averageMark)
+
+
+
+
+// ---- Verwachte uitkomst: 6.642857142857143
+
+
+/* 2b: Omschrijven tot een herbruikbare functie */
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+let averageMark2 = 0;
+function averageGrade (averageCalculate){
+    for (let i = 0; i < averageCalculate.length; i++) {
+        if ([i] < averageCalculate.length -1 ) {
+            averageMark2 = averageMark2 + averageCalculate[i]
+        } else {
```

Comment:
Met het vorige script en uitleg (zie 2a), wordt deze functie ook een stuk kleiner. In de return kan je deze dan terug geven, in plaats van in de console log. 

Let dan ook op: het variabel wordt dan binnen de functie geplaatst.